### PR TITLE
Make wheel install site-packages detection robust.

### LINKED ISSUE #1861: Pex fails to find RECORD for python-certifi-win32 1.6.1
Looks like:
```
$ pex python-certifi-win32==1.6.1
Traceback (most recent call last):
  File "/home/jsirois/.venv/pex/bin/pex", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/jsirois/.venv/pex/lib/python3.11/site-packages/pex/bin/pex.py", line 775, in main
    catch(
    ^^^^^^
  File "/home/jsirois/.venv/pex/lib/python3.11/site-packages/pex/result.py", line 103, in catch
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/jsirois/.venv/pex/lib/python3.11/site-packages/pex/bin/pex.py", line 798, in do_main
    pex_builder = build_pex(
                  ^^^^^^^^^^
  File "/home/jsirois/.venv/pex/lib/python3.11/site-packages/pex/bin/pex.py", line 681, in build_pex
    result = resolve(
             ^^^^^^^^
  File "/home/jsirois/.venv/pex/lib/python3.11/site-packages/pex/resolver.py", line 948, in resolve
    build_and_install_request.install_distributions(
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jsirois/.venv/pex/lib/python3.11/site-packages/pex/resolver.py", line 714, in install_distributions
    for install_result in execute_parallel(
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jsirois/.venv/pex/lib/python3.11/site-packages/pex/jobs.py", line 541, in execute_parallel
    yield spawn_result.spawned_job.await_result()
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jsirois/.venv/pex/lib/python3.11/site-packages/pex/jobs.py", line 220, in await_result
    job.wait()
    ^^^^^^^^^^
  File "/home/jsirois/.venv/pex/lib/python3.11/site-packages/pex/jobs.py", line 81, in wait
    self._finalize_job()
    ^^^^^^^^^^^^^^^^^^^^
  File "/home/jsirois/.venv/pex/lib/python3.11/site-packages/pex/jobs.py", line 139, in _finalize_job
    self._finalizer(self._process.returncode)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jsirois/.venv/pex/lib/python3.11/site-packages/pex/pip/tool.py", line 711, in fixup_install
    record = Record.from_prefix_install(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jsirois/.venv/pex/lib/python3.11/site-packages/pex/pep_376.py", line 445, in from_prefix_install
    raise RecordNotFoundError(
    ^^^^^^^^^^^^^^^^^^^^^^^^^^
pex.pep_376.RecordNotFoundError: Could not find the installation RECORD for python-certifi-win32 1.6.1 under /home/jsirois/.pex/installed_wheels/508fd4fb1730cad2d9dada061df737650c8cfaa205d64657faa4cc6a55384402/python_certifi_win32-1.6.1-py2.py3-none-any.whl.353db331daf74436a74d93cf324aa9a9
```

### PR DESCRIPTION
**Pull request by jsirois**
We already had robust and well tested code for this; so just re-plumb a
bit and use it.

Fixes #1861

### CODE REVIEW

**Review by Eric-Arellano — APPROVED** [review]

File: pex/pep_376.py [hunk 1]
```diff
@@ -15,19 +15,12 @@
 from fileinput import FileInput
 
 from pex import dist_metadata, hashing
-from pex.common import (
-    filter_pyc_dirs,
-    filter_pyc_files,
-    find_site_packages,
```
Comment (benjyw) [inline pex/pep_376.py:21]: Can we delete `find_site_packages` ?

**Review by benjyw — APPROVED** [review]

File: pex/pep_376.py (hunk 1 above)
Comment (jsirois) [reply pex/pep_376.py:21]: Nope, vendoring uses it. I'll follow up with inlining if the same new function extracted in this PR can't be used. Using Pex code during vendoring is very fragile! As such, that seemed too much to take on in this PR.

Comment (benjyw) [reply pex/pep_376.py:21]: Reasonable! Although probably worth at least documenting why both functions exist, if they do continue to exist.

Comment (jsirois) [reply pex/pep_376.py:21]: They won't. It'll be inline or else kill and use new.

Comment (jsirois) [reply pex/pep_376.py:21]: https://github.com/pantsbuild/pex/pull/1865

**Pull request merged**